include/cantera/oneD/refine.h

```diff
@@ -69,51 +123,142 @@ class Domain1D;
         return m_gridmin;
     }
 
-    //! Determine locations in the grid that need additional grid points.
-    //!
-    //! @returns The number of new grid points needed (size of #m_loc)
+    /**
+     * Determine locations in the grid that need additional grid points and
+     * updates the internal state of the Refiner with this information.
+     *
+     * @param[in] n  Number of grid points
+     * @param[in] z  Point to array of grid points
+     * @param[in] x  Pointer to solution vector
+     *
+     * @returns The number of new grid points needed (size of #m_insertPts)
+     */
     int analyze(size_t n, const double* z, const double* x);
+
+    /**
+     * Constructs a new grid based on refinement locations determined by the analyze
+     * method.
+     *
+     * This function generates a new grid by inserting additional points into the
+     * current grid at locations where the `analyze` method has identified a need for
+     * refinement. The new grid points are placed midway between existing points deemed
+     * necessary for increased resolution. If no refinement is needed, the original
+     * grid is copied directly.
+     *
+     * @param[in] n The number of points in the original grid array `z`.
+     * @param[in] z Pointer to the array of original grid points.
+     * @param[in] nn The maximum number of points that the new grid array `znew` can hold.
+     * @param[out] znew Pointer to the array where the new grid points will be stored.
+     *
+     * @return The function returns 0 upon successful creation of the new grid. Throws
+     *         an exception if the provided output array size is insufficient to hold
+     *         the new grid.
+     *
+     * @deprecated Starting in %Cantera 3.1, this function is unused. To be removed
+     *             after %Cantera 3.1.
+     */
     int getNewGrid(int n, const double* z, int nn, double* znew);
+
+    /**
+     * Returns the number of new grid points that were needed.
+     */
```

Comment:
For a function with a single-line docstring, you can save some vertical real estate by just writing
```suggestion

    //! Returns the number of new grid points that were needed.
```